<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Evolutionary models</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 54px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h2 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h3 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h4 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h5 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h6 {
  padding-top: 59px;
  margin-top: -59px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Code</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-file-alt"></span>
     
    Sequence files
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="sequence-files.html">Command line</a>
    </li>
    <li>
      <a href="visualise.html">Visualising and editing</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-binoculars"></span>
     
    Alignment
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="muscle.html">Muscle</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-code-branch"></span>
     
    Phylogenetics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="phylo-intro.html">Introduction</a>
    </li>
    <li>
      <a href="evolutionary-models.html">Evolutionary models</a>
    </li>
    <li>
      <a href="mrbayes.html">MrBayes</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Evolutionary models</h1>

</div>


<p><a href="https://shields.io/"><img src="https://img.shields.io/badge/Siobhon%20Egan-August%202019-%3CCOLOR%3E.svg" alt="Generic badge" /></a></p>
<p>An excellent summary of the different models (listed below) is directly sourced from Evolution and Geonomics Workshop (<a href="http://evomics.org/resources/substitution-models/nucleotide-substitution-models/">link here</a>).</p>
<p>Highly recommend you check out the rest of the website for some great resources and detailed information &amp; tutorials.</p>
<p>The use of maximum likelihood (ML) algorithms in developing phylogenetic hypotheses requires a model of evolution. The frequently used General Time Reversible (GTR) family of nested models encompasses 64 models with different combinations of parameters for DNA site substitution. The models are listed here from the least complex to the most parameter rich.</p>
<p>Some common programs to carry out phylogenetic analysis include:</p>
<ul>
<li><a href="http://nbisweden.github.io/MrBayes/">MrBayes</a></li>
<li><a href="https://paup.phylosolutions.com/">PAUP*</a></li>
<li><a href="http://www.beast2.org">BEAST v2</a></li>
<li><a href="http://www.atgc-montpellier.fr/phyml/">PhyML v3</a></li>
<li><a href="http://evomics.org/resources/software/molecular-evolution-software/paml/">PAML</a></li>
<li><a href="https://sco.h-its.org/exelixis/software.html">RAxML</a> generally only utilises the GTR model, although you can use JC and K80 models for DNA partitions.</li>
</ul>
<p>The below also include the settings if the model is available in these programs.</p>
<p><a href="http://phylobotanist.blogspot.com/2016/06/implementation-of-substitution-models.html" class="uri">http://phylobotanist.blogspot.com/2016/06/implementation-of-substitution-models.html</a></p>
<p><strong>Jukes-Cantor (JC/JC69)</strong> equal base frequencies, all substitutions equally likely (Jukes and Cantor 1969)</p>
<ul>
<li>[x] MrBayes <code>nst=1</code></li>
<li>[x] PAUP <code>aaaaaa</code></li>
<li>[x] PAML <code>aaaaaa</code></li>
<li>[x] BEAST</li>
<li>[x] PhyML</li>
</ul>
<p><strong>Felsenstein 1981 (F81)</strong> variable base frequencies, all substitutions equally likely (Felsenstein 1981)</p>
<ul>
<li>[x] MrBayes <code>nst=1</code></li>
<li>[x] PAUP <code>aaaaaa</code></li>
<li>[x] PAML <code>aaaaaa</code></li>
<li>[x] BEAST (check)</li>
<li>[x] PhyML</li>
</ul>
<p><strong>Kimura 2-parameter (K80/K2)</strong> equal base frequencies, one transition rate and one transversion rate (Kimura 1980)</p>
<ul>
<li>[x] MrBayes <code>nst=2</code></li>
<li>[x] PAUP <code>abaaba</code></li>
<li>[x] PAML <code>abbbba</code></li>
<li>[x] BEAST</li>
<li>[x] PhyML</li>
</ul>
<p><strong>Hasegawa-Kishino-Yano (HKY)</strong> variable base frequencies, one transition rate and one transversion rate (Hasegawa et. al. 1985). Note this model is very similar to K80 however allows for variable base frequencies. It is also commonly the default model in many programs.</p>
<ul>
<li>[x] MrBayes <code>nst=2</code></li>
<li>[x] PAUP <code>abaaba</code></li>
<li>[x] PAML <code>abbbba</code></li>
<li>[x] BEAST</li>
<li>[x] PhyML</li>
</ul>
<p><strong>Tamura-Nei (TrN/TN93)</strong> variable base frequencies, equal transversion rates, variable transition rates (Tamura Nei 1993)</p>
<ul>
<li>[ ] MrBayes</li>
<li>[x] PAUP <code>abaaea</code></li>
<li>[x] PAML <code>abbbbf</code></li>
<li>[x] BEAST</li>
<li>[x] PhyML</li>
</ul>
<p><strong>Kimura 3-parameter (K3P)</strong> variable base frequencies, equal transition rates, two transversion rates (Kimura 1981)</p>
<ul>
<li>[ ] MrBayes</li>
<li>[x] PAUP (<code>abccba</code>)</li>
<li>[x] PAML (<code>abccba</code>)</li>
<li>[ ] BEAST</li>
<li>[ ] PhyML</li>
</ul>
<p><strong>transition model (TIM)</strong> variable base frequencies, variable transition rates, two transversion rates</p>
<ul>
<li>[ ] MrBayes</li>
<li>[x] PAUP <code>abccea</code></li>
<li>[x] PAML <code>abccbe</code></li>
<li>[x] BEAST</li>
<li>[ ] RAxML (check)</li>
<li>[ ] PhyML</li>
</ul>
<p><strong>transversion model (TVM)</strong> variable base frequencies, variable transversion rates, transition rates equal</p>
<ul>
<li>[ ] MrBayes</li>
<li>[x] PAUP (<code>abcdbe</code>)</li>
<li>[x] PAML (<code>abcdea</code>)</li>
<li>[x] BEAST</li>
<li>[ ] PhyML</li>
</ul>
<p><strong>symmetrical model (SYM)</strong> equal base frequencies, symmetrical substitution matrix (A to T = T to A) (Zharkikh 1994)</p>
<ul>
<li>[x] MrBayes <code>nst=6</code></li>
<li>[x] PAUP <code>abcdef</code></li>
<li>[x] PAML <code>abcdef</code></li>
<li>[x] BEAST</li>
<li>[ ] PhyML</li>
</ul>
<p><strong>general time reversible (GTR)</strong> variable base frequencies, symmetrical substitution matrix (e.g., Lanave et al. 1984, Tavare 1986, Rodriguez et. al. 1990)</p>
<ul>
<li>[x] MrBayes <code>nst=6</code></li>
<li>[x] PAUP <code>abcdef</code></li>
<li>[x] PAML <code>abcdef</code></li>
<li>[x] BEAST</li>
<li>[x] PhyML</li>
</ul>
<div id="rate-variation" class="section level3">
<h3>Rate variation</h3>
<p>In addition to models describing the rates of change from one nucleotide to another, there are models to describe rate variation among sites in a sequence. The following are the two most commonly used models and are generally available across all platforms and programs.</p>
<p><strong>gamma distribution (G)</strong> gamma distributed rate variation among sites</p>
<p><strong>proportion of invariable sites (I)</strong>* extent of static, unchanging sites in a dataset</p>
<hr />
</div>
<div id="programs" class="section level1">
<h1>Programs</h1>
<p>Some well resourced programs to carry out model selection include: - <a href="https://www.megasoftware.net/">MEGA</a> - <a href="http://www.atgc-montpellier.fr/phyml/">PhyML</a> - <a href="https://github.com/ddarriba/jmodeltest2">jmodeltest</a></p>
<p>Depending on the program most will produce two sets of ‘scores’ to assess the models. the Akaike information criterion (AIC) and Bayesian information criterion (BIC). The lower the score the more support there is for a particular model. There are different reasons for choosing the best model based on the AIC or BIC scores. Generally the BIC score is used in most cases, however if in doubt its always best to do some more research to understand why you chose that value. Some references for more information on these criterion available here <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2925852/">Luo A. <em>et al.</em> 2010</a>; <a href="https://www.annualreviews.org/doi/pdf/10.1146/annurev.ecolsys.36.102003.152633">Sullivan J. and Joyce P. 2005</a>,<a href="https://besjournals.onlinelibrary.wiley.com/doi/full/10.1111/2041-210X.12541">Brewer M. J. <em>et al.</em> 2016</a>.</p>
<p>Example output from MEGA7 using Model Selection feature <img src="MEGA7_ModelSelection.png" alt="MEGA7_ModelSelection" /></p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
