<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>MrBayes</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 54px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h2 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h3 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h4 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h5 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h6 {
  padding-top: 59px;
  margin-top: -59px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->





<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Code</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="sequence-files.html">
    <span class="fa fa-file-alt"></span>
     
    Sequence files
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-binoculars"></span>
     
    Alignment
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="muscle.html">Muscle</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-code-branch"></span>
     
    Phylogenetics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="phylo-intro.html">Introduction</a>
    </li>
    <li>
      <a href="evolutionary-models.html">Evolutionary models</a>
    </li>
    <li>
      <a href="mrbayes.html">MrBayes</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">MrBayes</h1>

</div>


<p>Notes for command line MrBayes</p>
<p><strong>Useful resources</strong> - <a href="http://mrbayes.sourceforge.net/mb3.2_manual.pdf">Manual v3.2</a> - <a href="http://mrbayes.sourceforge.net/wiki/index.php?title=Tutorial&amp;oldid=5239">Tutorial</a> - <a href="http://people.binf.ku.dk/rasmus/courses/Spring06/Chapter7-MrBayes.pdf">John P. Huelsenbeck and Fredrik Ronquist, Bayesian Analysis of Molecular Evolution Using MrBayes</a></p>
<div id="step-1---convert-your-sequence-file-to-nexus-format" class="section level3">
<h3>Step 1 - Convert your sequence file to nexus format</h3>
<p>You will need to convert your aligned sequence file to a nexus file first. Although this can commonly be done in most sequence analysis programs (e.g. geneious) using the <code>export as...</code> make sure you check the file formatting by opening the <code>nexus</code> file in a text editor program (e.g. atom).</p>
<p>I find a more reliable way is to use a web program to convert the fasta file to a nexus file - such as one available <a href="http://sequenceconversion.bugaco.com/converter/biology/sequences/fasta_to_nexus.php">here</a>. Make sure the sequence names do not contain any of the following; hyphens <code>-</code>, commas<code>,</code>, semicolcons <code>:</code>, exclamation marks <code>!</code>. Underscores <code>_</code> and periods <code>.</code> are okay.</p>
<blockquote>
<p>For details on how to download MrBayes see the <a href="http://mrbayes.sourceforge.net/mb3.2_manual.pdf">manual section 1.2</a></p>
</blockquote>
</div>
<div id="step-2---activate-mrbayes" class="section level3">
<h3>Step 2 - Activate MrBayes</h3>
<p>Type <code>mb</code> to activate MrBayes</p>
<pre><code>execute AlignedSeqs.nexus</code></pre>
<p>This will check will ensure your data is formatting and read correctly by the program. Any issues at this point are most likely due to the syntax and formatting of your nexus file. Note: you may need to also include the path to the file if you are not already in the correct directory. Generic Shell commands (e.g. <code>ls</code> <code>cd</code>) do not work in the MrBayes environment.</p>
</div>
<div id="step-3---set-the-evolutionary-model" class="section level3">
<h3>Step 3 - Set the evolutionary model</h3>
<p>This is where you start to set the parameters for your analysis. You will likely need to do some background work to figure out which is the correct model for your data - <a href="/Users/siobhonegan/Documents/Uni/PhD/code/notes-cheatsheets/Evolutionary_model_selection.md">see here</a></p>
<p>If your data are not DNA or RNA, if you want to invoke a different model, or if you want to use non-default priors, refer to the MrBayes manual and Appendix for more help.</p>
<div class="figure">
<img src="MrBayesv3_DNA_EvolutionaryModel.png" alt="DNA_EvolModel" />
<p class="caption">DNA_EvolModel</p>
</div>
<p><em>DNA models support by MrBayes v3 (MrBayes Manual p164)</em></p>
<p><strong>Settings</strong></p>
<table>
<thead>
<tr class="header">
<th>Parameter</th>
<th align="center">Options</th>
<th align="right">Current setting</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Nucmodel</td>
<td align="center">4by4/Doublet/Codon/Protein</td>
<td align="right">4by4</td>
</tr>
<tr class="even">
<td>Nst</td>
<td align="center">1(JC/F81)/2(HKY)/6(GTR)/Mixed</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td>Code</td>
<td align="center">Universal/Vertmt/Mycoplasma/Yeast/Ciliates/Metmt</td>
<td align="right">Universal</td>
</tr>
<tr class="even">
<td>Ploidy</td>
<td align="center">Haploid/Dilpoid/Zlinked</td>
<td align="right">Diploid</td>
</tr>
<tr class="odd">
<td>Rates</td>
<td align="center">Equal/Gamma/Propinv/Invgamma/Adgamma</td>
<td align="right">Equal</td>
</tr>
<tr class="even">
<td>Ngammacat</td>
<td align="center">number</td>
<td align="right">4</td>
</tr>
<tr class="odd">
<td>Nbetacat</td>
<td align="center">number</td>
<td align="right">5</td>
</tr>
<tr class="even">
<td>Omegavar</td>
<td align="center">Equal/Ny93/M3</td>
<td align="right">Equal</td>
</tr>
<tr class="odd">
<td>Covarion</td>
<td align="center">No/Yes</td>
<td align="right">No</td>
</tr>
<tr class="even">
<td>Coding</td>
<td align="center">All/Variable/Noabsencesites/Nopresencesites</td>
<td align="right">All</td>
</tr>
<tr class="odd">
<td>Parsmodel</td>
<td align="center">No/Yes</td>
<td align="right">No</td>
</tr>
</tbody>
</table>
<p>To set a GTR + I + G model execute the following:</p>
<pre><code>lset nst=6 rates=invgamma</code></pre>
</div>
<div id="step-4-start-sampling" class="section level3">
<h3>Step 4 start sampling</h3>
<pre><code>mcmc ngen=1000000 samplefreq=500 printfreq=500 diagnfreq=5000</code></pre>
<p>If the standard deviation of split frequencies is below 0.01 the specific number of generations, stop the run by answering no when the program asks <code>Continue the analysis? (yes/no)</code>. Otherwise, keep adding generations until the value falls below <code>0.01</code>. If you are interested mainly in the well-supported parts of the tree, a standard deviation below <code>0.05</code> may be adequate.</p>
</div>
<div id="get-output" class="section level3">
<h3>Get output</h3>
<p>Type <code>sump</code> to summarize the parameter values using the same burn-in as the diagnostics in the mcmc command. The program will output a table with summaries of the samples of the substitution model parameters, including the mean, mode, and 95 % credibility interval (region of Highest Posterior Density, HPD) of each parameter. Make sure that the potential scale reduction factor (PSRF) is reasonably close to 1.0 for all parameters; if not, you need to run the analysis longer.</p>
<p>Summarize the trees using the same burn-in as the mcmc command by typing <code>sumt</code>. The program will output a cladogram with the posterior probabilities for each split and a phylogram with mean branch lengths. Both trees will also be printed to a file that can be read by FigTree and other tree-drawing programs, such as TreeView and Mesquite.</p>
<p>It does not have to be more complicated than this; however, as you get more proficient you will probably want to know more about what is happening behind the scenes. The rest of this section explains each of the steps in more detail and introduces you to all the implicit assumptions you are making and the machinery that MrBayes uses in order to perform your analysis.</p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
