<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Phylogenetics Introduction</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 54px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h2 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h3 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h4 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h5 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h6 {
  padding-top: 59px;
  margin-top: -59px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->





<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Code</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="sequence-files.html">
    <span class="fa fa-file-alt"></span>
     
    Sequence files
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-binoculars"></span>
     
    Alignment
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="muscle.html">Muscle</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-code-branch"></span>
     
    Phylogenetics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="phylo-intro.html">Introduction</a>
    </li>
    <li>
      <a href="evolutionary-models.html">Evolutionary models</a>
    </li>
    <li>
      <a href="mrbayes.html">MrBayes</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Phylogenetics Introduction</h1>

</div>


<p>There are four popular molecular phylogenetic methods. The following information is a summary of information provided by material from the Australian National Postgraduate Training Workshop in Systematics support by the <a href="https://www.sasb.org.au">Society of Australian Systematic Biologists</a>, <a href="http://www.environment.gov.au/science/abrs">The Australian Biological Resources Study</a> and <a href="https://www.adelaide.edu.au/environment/acebb/">Australian Centre for Evolutionary Biology and Biodiversity</a> <em>and</em> <a href="https://github.com/simon-ho/SydneyPhyloWorkshop">Sydney Phylo Workshop</a> by Simon Ho.</p>
<p><strong>1. Maximum parsimony</strong> Identifies the tree topology that can explain the sequence data, using the smallest number of inferred substitution events. It is commonly used for morphological data, and now days is rarely used for analysing genetic data. It cannot estimate evolutionary rates or timescales and does not correct for multiple substitutions at the same site which leads to what is known as “long-branch attraction”. <strong>2. Distance-based methods</strong> Clustering algorithms - Unweighted Pair Group Method with Arithmetic Mean (UPGMA) - Neighbour joining (NJ) Tree searching using optimality criteria - Minimum evolution - Least-squares inference</p>
<p><em>Pros</em> - Able to take into account multiple substitutions and long-branch attraction - Useful for analysing very large data sets with a lot of taxa (1000’s +) <em>Cons</em> - Does not use all information in alignment - Loss of information in pairwise comparisons - Unable to implement sophisticated evolutionary models</p>
<p><strong>3. Maximum likelihood (ML)</strong> Search through the space of possible trees and parameter values Calculate the likelihood for these Find best tree and model parameter values Multivariate optimisation</p>
<p><em>Pros</em> - Rigorous statistical method - Deals with multiple substitutions and long-branch attraction - Highly robust to violations of assumptions <em>Cons</em> - Not feasible to implement very parameter-rich models - Searching tree space can be difficult</p>
<p><em>ML Analysis</em> * RAxML = Randomized Axelerated Maximum Likelihood. Rapid bootstrapping and can run sequentially or in parallel. * PhyML * MEGA * PAML * IQ-Tree</p>
<p><strong>4. Bayesian inference</strong> Bayesian phylogenetic analysis was developed in the mid 1990s, and it now one of the most widely used methods for molecular phylogenetics.</p>
<p>Key features of Bayesian paradigm - Contrast with frequentist statistics (likelihood) - Parameters have distributions - Before the data are observed, each parameter has a prior distribution - The likelihood of the data is computed - The prior distribution is combined (updated) with the likelihood to yield the posterior distribution</p>
<p><em>Priors</em> - Priors are chosen in the form of probability distributions - Reflect our prior expectations (and uncertainty) about values of parameters (without knowledge of the data) * Past observations * Personal beliefs * Use of a biological model - Uninformative priors</p>
<p><em>‘Prior options’</em> 1. Use a flat prior for tree topology (MrBayes) - All trees have equal probability - Also need a prior for branch lengths or node times</p>
<ol start="2" style="list-style-type: decimal">
<li>Use a biological model to generate prior distribution (BEAST and MrBayes)</li>
</ol>
<ul>
<li>Among species: speciation model</li>
<li>Within species: coalescent model</li>
</ul>
<p><em>Markov Chain Monte Carlo Sampling</em> It is impossible to obtain the posterior directly. Instead, the posterior can be estimated using Markov chain Monte Carlo simulation. This is usually done using the Metropolis-Hastings algorithm. There is a wealth of literature available for more information on this topic. It can get super technical - we’ve just covered the basics.</p>
<p><em>Pros</em> - Able to implement highly parameterised models - Estimating tree uncertainty is straightforward - Can only do this indirectly in likelihood (via bootstrapping) - Posterior probabilities have an intuitive interpretation - Can incorporate independent information (in the prior)</p>
<p><em>Popular software and programs for Bayesian analysis</em> - MrBayes * Primarily designed for species-level data * Simultaneous estimation of tree and node times * Range of clock models * Range of tree priors * Multiple chains and MCMC diagnostics - RevBayes * Uses its own R-like language, Rev * Interactive construction of graphical model * Flexible and can be used for simulation and inference * Ongoing development - BEAST1 = Bayesian Evolutionary Analysis by Sampling Trees * Analyse population- or species-level data * Simultaneous estimation of tree and node times * Range of clock models * Range of tree priors and demographic models - BEAST2 * Re-write of BEAST to increase modularity * Users can extend BEAST by adding packages * Additional tree priors not available in BEAST 1 * Capacity to perform simulations * For a comparison of BEAST 1 and 2: www.beast2.org/beast-features</p>
<div id="bootstrapping" class="section level5">
<h5>Bootstrapping</h5>
<p>Felsenstein (1985) Evolution 39(4):783-791 (<a href="https://www.ncbi.nlm.nih.gov/pubmed/28561359">Pub med</a>) Bootstrapping provides a confidence interval that contains the phylogeny that would be estimated from repeated sampling of many characters from the underlying set of all characters. Bootstrap values are measures of repeatability. High when the data set is large. Not meaningful when analysing genome-scale data.</p>
<hr />
</div>
<div id="useful-resources-for-molecular-phylogenetics" class="section level2">
<h2>Useful resources for molecular Phylogenetics</h2>
<ul>
<li><a href="https://github.com/simon-ho/SydneyPhyloWorkshop">Sydney Phylo Workshop</a> by Simon Ho provides some excellent lecture notes and workshops with dataset</li>
<li><a href="https://github.com/Phylogenetics-Brown-BIOL1425/phylogeneticbiology">Phylogenetic Biology</a> by Casey Dunn at Brown University is a great source of lecture and practical material. It also has a an extensive list of extra resources and useful links to reading material</li>
<li><a href="https://github.com/revbayes">RevBayes</a> github page has a wealth of information specifically around the software but also includes some general phylogenetics material as well. Tutorial pages available <a href="https://revbayes.github.io/tutorials/">here</a>, a example of workshop page on molecular phylogenetics and fossil data available <a href="http://phyloworks.org/revbayes-workshop2017/">here</a></li>
<li><a href="https://taming-the-beast.org/tutorials/Introduction-to-BEAST2/">Taming the BEAST</a> workshops around the use of BEAST software for Phylogenetics</li>
<li><a href="https://github.com/thednainus/Bayesian_tutorial">Bayesian MCMC phylogenetics tutorial in R</a> - for those wanting an R interface for their Phylogenetics</li>
<li><a href="https://github.com/mmatschiner/tutorials">Collection of tutorials on phylogenetics</a> by Michael Matschiner. This goes through the workflow from sequence analysis through to dating and more</li>
</ul>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
