<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Sequence file formats and basics</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 54px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h2 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h3 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h4 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h5 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h6 {
  padding-top: 59px;
  margin-top: -59px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Code</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-file-alt"></span>
     
    Sequence files
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="sequence-files.html">Command line</a>
    </li>
    <li>
      <a href="visualise.html">Visualising and editing</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-binoculars"></span>
     
    Alignment
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="muscle.html">Muscle</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-code-branch"></span>
     
    Phylogenetics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="phylo-intro.html">Introduction</a>
    </li>
    <li>
      <a href="evolutionary-models.html">Evolutionary models</a>
    </li>
    <li>
      <a href="mrbayes.html">MrBayes</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Sequence file formats and basics</h1>

</div>


<p><a href="https://shields.io/"><img src="https://img.shields.io/badge/Siobhon%20Egan-August%202019-%3CCOLOR%3E.svg" alt="Generic badge" /></a></p>
<div id="basic-information-and-links" class="section level2">
<h2>Basic information and links</h2>
<p>This section first starts off with some quick useful commands that will be helpful to assess and manipulate your fasta and fastq file. They can be done locally in your command line and don’t require any clumsy dependancies.</p>
<p>Credit to some helpful inspiration</p>
<ul>
<li><a href="http://bioinformatics.cvr.ac.uk/blog/short-command-lines-for-manipulation-fastq-and-fasta-sequence-files/">Bioinformatics I/O by Joseph Hughes</a></li>
<li><a href="https://github.com/stephenturner/oneliners#basic-awk--sed">Bioinformatic one-liners by Stephen Turner</a></li>
<li><a href="http://userweb.eng.gla.ac.uk/umer.ijaz/bioinformatics/linux.html">Linux command line exercises for NGS data processing by Umer Zeeshan Ijaz</a></li>
<li><a href="https://www.biostars.org/p/17680/">Biostars thread</a></li>
</ul>
<p>These examples simply use <code>sample.fa</code> or <code>sample.fq</code> or similar format to represent the file you want to investigate. Remember you will need to navigate to the the path of the file or ensure you are in the correct working directory.</p>
<p><strong>Convert a fastq file to a fasta file</strong></p>
<pre class="console"><code>sed &#39;/^@/!d;s//&gt;/;N&#39; sample.fq &gt; sample1.fa</code></pre>
<p><strong>Count number of sequences</strong></p>
<p>Fastq file</p>
<pre class="console"><code>grep -c &#39;^@&#39; sample.fq</code></pre>
<p>Fasta file</p>
<pre class="console"><code>grep -c &#39;&gt;&#39; sample.fasta</code></pre>
<p><strong>Get some basic info on your file(s)</strong></p>
<pre class="console"><code>perl -ne &#39;if(/^&gt;(\S+)/){print &quot;$1\n&quot;}&#39; sample.fa</code></pre>
<p>You may want to print them to a text file</p>
<pre class="console"><code>perl -ne &#39;if(/^&gt;(\S+)/){print &quot;$1\n&quot;}&#39; sample.fa &gt; sample_names.txt</code></pre>
<p>Get the sequence name and length for every individual sequence within a fasta file</p>
<pre class="console"><code>cat sample.fasta | awk &#39;$0 ~ &quot;&gt;&quot; {print c; c=0;printf substr($0,2,100) &quot;\t&quot;; } $0 !~ &quot;&gt;&quot; {c+=length($0);} END { print c; }&#39;</code></pre>
<p>Calculate the mean length of reads in a fastq file:</p>
<pre class="console"><code>awk &#39;NR%4==2{sum+=length($0)}END{print sum/(NR/4)}&#39; sample.fastq</code></pre>
<p>Add something to the end of all header lines</p>
<pre class="console"><code>sed &#39;s/&gt;.*/&amp;WHATEVERYOUWANT/&#39; file.fa &gt; outfile.fa</code></pre>
<p>Clean up a fasta file so that only the first column of the header is outputted</p>
<pre class="console"><code>awk &#39;{print $1}&#39; file.fa &gt; output.fa</code></pre>
<p>Remove dupilcate sequences from your fasta file</p>
<pre class="console"><code>sed -e &#39;/^&gt;/s/$/@/&#39; -e &#39;s/^&gt;/#/&#39; file.fasta | tr -d &#39;\n&#39; | tr &quot;#&quot; &quot;\n&quot; | tr &quot;@&quot; &quot;\t&quot; | sort -u -t $&#39;\t&#39; -f -k 2,2  | sed -e &#39;s/^/&gt;/&#39; -e &#39;s/\t/\n/&#39;</code></pre>
<p>Get A T (or U) G C counts for all sequences in a fasta file</p>
<pre class="console"><code>echo -e &quot;seq_id\tA\tU\tG\tC&quot;; while read line; do echo $line | grep &quot;&gt;&quot; | sed &#39;s/&gt;//g&#39;; for i in A U G C;do echo $line | grep -v &quot;&gt;&quot; | grep -o $i | wc -l | grep -v &quot;^0&quot;; done; done &lt; your_fasta_file.fa | paste - - - - -</code></pre>
<p>Split a multifasta in to separate files (with arbitrary names).</p>
<pre class="console"><code>splitfa(){
i=1;
while read line ; do
  if [ ${line:0:1} == &quot;&gt;&quot; ] ; then
    header=&quot;$line&quot;
    echo &quot;$header&quot; &gt;&gt; seq&quot;${i}&quot;.fasta
  else
    seq=&quot;$line&quot;
    echo &quot;$seq&quot; &gt;&gt; seq&quot;${i}&quot;.fasta
    ((i++))
  fi
done &lt; $1
}</code></pre>
<p><strong>Extract sequences by their ID</strong></p>
<p>Call just certain sequences of interest - in this case we are interested in sequences named <code>seq1</code>, <code>seq2</code> and <code>seq3</code> and put them in a file call <code>new.fasta</code></p>
<pre class="console"><code>perl -ne &#39;if(/^&gt;(\S+)/){$c=grep{/^$1$/}qw(seq1 seq2 seq3)}print if $c&#39; sample.fa &gt; new.fasta</code></pre>
<p>Say you have a text file with a list of sequences you want to further investigate you can use this text file (called <code>seqs_of_interest.txt</code>) to call out the sequences in the <code>sample.fasta</code> file and put them in a few file <code>seqs_of_interest.fasta</code></p>
<pre class="console"><code>perl -ne &#39;if(/^&gt;(\S+)/){$c=$i{$1}}$c?print:chomp;$i{$_}=1 if @ARGV&#39; seqs_of_interest.txt sample.fasta &gt; seqs_of_interest.fasta</code></pre>
<hr />
<hr />
</div>
<div id="useful-programs" class="section level1">
<h1>Useful programs</h1>
<p>There are a lot of programs out that. Some use the command line interface, there is also plenty of GUI and web-based programs too. Below are just two on the most useful ones that I like (I tend to like the command line interface ones, as they can easily be upscaled and used on your VM/supercomputer if needed).</p>
<div id="seqkit---a-cross-platform-and-ultrafast-toolkit-for-fasta-and-fastq-file-manipulation" class="section level2">
<h2>SeqKit - a cross platform and ultrafast toolkit for fasta and fastq file manipulation</h2>
<p>The above commands are good for some basic editing and information but if you want to do more with your fasta and fastq files your going to need some dedicated programs.</p>
<p><code>seqkit</code> is a super useful program for any bioinformatician follow <a href="https://github.com/shenwei356/seqkit">this link</a> for offical downloads and documenation.</p>
<p>We’ll run through the basic below but always refer to the link above for any issues and more details.</p>
<p><strong>Install (Linux-like systems)</strong></p>
<p>Just <a href="https://github.com/shenwei356/seqkit/releases">download</a> compressed executable file of your operating system. Navigate to the file in your command line and decompress it with <code>tar -zxvf *.tar.gz</code> command. * If you have root privilege simply copy it to <code>/usr/local/bin:</code> <code>sudo cp seqkit /usr/local/bin/`` * Or copy to anywhere in the environment variable PATH:</code>mkdir -p $HOME/bin/; cp seqkit $HOME/bin/`</p>
</div>
<div id="bioawk" class="section level2">
<h2>Bioawk</h2>
<p>This is a super useful addition to the awk range of commands specific to bioinformatics - see github page <a href="https://github.com/lh3/bioawk">here</a></p>
<p><strong>conda install</strong> Easy install using terminal, as long as you have conda installed <a href="https://anaconda.org/bioconda/bioawk" class="uri">https://anaconda.org/bioconda/bioawk</a></p>
<pre class="console"><code>conda install -c bioconda bioawk
conda install -c bioconda/label/cf201901 bioawk</code></pre>
<p><strong>Useful tutorials</strong></p>
<p>Detailed bioinformatics workbook from the <a href="https://gif.biotech.iastate.edu/">Genome Informatics Facility</a> - tutorial available <a href="https://isugenomics.github.io/bioinformatics-workbook/Appendix/bioawk-basics.html">here</a>, and github account <a href="https://github.com/ISUgenomics/bioinformatics-workbook">here</a>.</p>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
